edx_lint/pylint/unicode_check.py

```diff
@@ -58,7 +58,22 @@ def process_tokens(self, tokens):
             # If the whole module is __future__'d, then it's all fine.
             return
 
-        for tok_type, tok_text, start, _, _ in tokens:
+        previous_line = ""
+        current_line = ""
+        first_non_comment_line = 0
+        for index, (tok_type, tok_text, start, _, _) in enumerate(tokens):
+            # Use new line characters to build previous line
+            if first_non_comment_line == 0 and tok_type != 53: # 53 is token.COMMENT
```

Comment:
Pylint has an improved tokenizer that also provides comments: https://kite.com/python/docs/pylint.utils.tokenize

Reply:
The variable is called `first_non_comment_line`, but isn't a line or line number, it's a token number.  The first comment will be token 0, so the initialization to 0 seems wrong.  Use `None` instead.

Reply:
`first_non_comment_line` isn't a line or line number, it's a token index.  Tokens start at 0, so the init value should be `None`, not 0.

Reply:
i believe token.COMMENT is only defined for python 3 and was gave me an error when I tried to use it in the code.  
however, tok_type was 53 and token.tok_name[53] was 'COMMENT'

Reply:
hmm..well I'm iterating over tokens, not lines, but comments at the beginning of the file are treated as a single token.  
i guess i could call this variable first_non_comment_token, but I think the intent might be clearer if I call it first_non_comment_line, since if I have a string on the first non-comment line it is a docstring?  
I'm open to renaming it though